### PR TITLE
add normalization to LionSolver

### PR DESCRIPTION
**Pull request by cshoebridge**
Zak mentioned that it is often beneficial to normalize inputs.
This is done using global dataset statistics, so shouldn't be the job of a LIONModel, who has no knowledge of the dataset, rather the job of the LIONSolver who does.

Added functions to optionally enable normalization and logic in existing solvers to leverage this.

### CODE REVIEW

File: LION/optimizers/LIONsolver.py [hunk 1]
```diff
@@ -45,6 +46,19 @@ class SolverParams(LIONParameter):
     def __init__(self):
         super().__init__()
 
+def normalize_input(func):
+    def wrapper(self, *inputs):
+        if self.do_normalize:
+            print(inputs)
```
Comment (AnderBiguri) [inline LION/optimizers/LIONsolver.py:52]: Lets not print always. This is why self.verbose exist in LIONSolver, for example.

**Comment by AnderBiguri** [comment]
Great idea!
I wonder however if we should pre-do this. Iterating over the entire dataset can be super long, so maybe this should be in each DataSet rather than in LIONSolver? 

Unsure, actually, so maybe a thing worth discussing.

**Comment by AnderBiguri** [comment]
On second thought, even if it was in the datasets we provide, it would also need to be in the solver to be able to support generic datasets, so it makes sense

File: LION/optimizers/LIONsolver.py (hunk 1 above)
Comment (cshoebridge) [reply LION/optimizers/LIONsolver.py:52]: don't think it ever needs printing, that was for debugging 🤦 
fixed.

**Comment by AnderBiguri** [comment]
This has been tested I assume? happy to merge?

**Comment by cshoebridge** [comment]
yes good idea, could be worth looking at, though the inner working of the datasets is out of my realm of expertise

**Comment by cshoebridge** [comment]
and yes, tested, as set up it doesn't break anything else, it's entirely optional.

**Pull request merged**